plugins/emmc/emmc.quirk

```diff
@@ -0,0 +1,9 @@
+#Western Digital
+[EMMC\NAME_DI4064]

```

Comment:
I'm worried about other hardware accidentally matching this -- would the other GUID be more precise? e.g. `EMMC\MAN_%manfid%&OEM_%oemid%&NAME_%name%`

Reply:
(you can see the possible GUIDs to match in the plugins/emmc/README.md file)

Reply:
I would hardly believe.  For many years now with Sandisk - I am unaware of any conflicting product names.
Still, better safe than sorry - will update in my next spin.
Thanks.

Reply:
Ohh, i was worried about a less scrupulous vendor *naming* their device `DI4064` just to get the EmmcBlockSize quirk entry set...

Reply:
As of now, no other vendor quirked the ff chunk size, so we are alone on this.
And should anyone will want to - they can use whatever GUID they want - just the product name or other.
So unless anyone objects, I can leave it as it is for now.

Will wait for some more comments, the re-test and upload a new version next Monday.
Thanks,
Avri

Reply:
> So unless anyone objects

Can I object? Can you use the bogus value for oemid?

Reply:
> DI4064

I'm not going to veto it -- but it does seem overly fragile in my experience.

Reply:
If it should apply for all device, I would think it would be good to include manfid, but not sure if oemid is necessary - that field implies further subdivision if I understand the purpose of the filed correctly.

Reply:
Mirroring an off-list discussion with the product manager: "...

Then let's stick to the original proposal. Thanks.

Richard.

On Thu, 19 Oct 2023 at 15:57, Ranjini Durai <[Ranjini.Durai@wdc.com](mailto:Ranjini.Durai@wdc.com)> wrote:
Hi Avri,

It's very unlikely that other vendors/product use the same product name .

Thanks,
Ranjini

________________________________________
From: Avri Altman <[Avri.Altman@wdc.com](mailto:Avri.Altman@wdc.com)>
Sent: Thursday, October 19, 2023, 20:05
To: Ranjini Durai <[Ranjini.Durai@wdc.com](mailto:Ranjini.Durai@wdc.com)>
Cc: Avi Shchislowski <[Avi.Shchislowski@wdc.com](mailto:Avi.Shchislowski@wdc.com)>; Keren Gonen <[Keren.Gonen@wdc.com](mailto:Keren.Gonen@wdc.com)>; Richard Hughes <[hughsient@gmail.com](mailto:hughsient@gmail.com)>
Subject: RE: [fwupd/fwupd] An eMMC plugin changes (PR #6273)

Ranjini hi,
While reviewing the fwupd quirk patches, Richard – the LVFS maintainer (cc’ed),
has raised a concern about designating the quirk using just the product name.
 
How likely/unlikely is that some other vendor will have product name identical to ours?
 
Thanks,
Avri
..."

Reply:
My understanding is that the quirks are apply via a <guid,key,value> tuples.
Should some other vendor wants to apply whatever quirk, even the ffu-write-block-size, they can use whatever guid they choose, not necessarily the product name - which works for us.
Am I missing something here?